johnsonandjohnson/Bodiless-JS
# feat(organisms): Burger Menu

## Changes
- New component Burger Menu added into bodiless-organisms
- header uses new design API
- Added Burger Menu to the site header (visible only on mobile screens)
- Added new page /burger-menu/ with burger-menu example
- Basic Documentation for the Burger Menu
- updated package* files with new dependency - negomi/react-burger-menu
- Submenus in Burger Menu implemented as accordions
- If menu item with submenu is a link, than "Overview" link added to accordion body. "Overview" text is configurable by site-builder
- Updates for Main Menu: new HOC's: asMainMenu, asEditableMainMenu
## Technical Details

- Extended the site tailwind.config
- Burger Menu API matches Main Menu API
- new HOC - asStatic sets the component and its children to read-only mode
- fix for List, preventing unwanted properties to appear into static mode
- Standardized API for applying responsive classes between
- withEditableTitle HOC: Allow injection of asEditable to support different kinds of editors

## Test Instructions

1. Add to site's Main Menu next items:
  - Item with link and with submenu
  - Item without link and with submenu
  - Item with link and without submenu
  - Item without link and without submenu
2. Resize the browser screen to small size
3. Check that burger menu button is visible
4. Open the burger menu
5. Check that close button works as expected
6. Check that items with submenu rendered as accordionns
7. Check that item with link and with submenu has an "Overview" link in the accordion body

## References:

- https://www.npmjs.com/package/react-burger-menu

## Related Issues
